### PR TITLE
Fix failing test that checks for runtime loaded assembly when running in serial

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose
attempt fixing test for linux

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. Use N/A to indicate that the changes in this pull request do not apply to Release Notes. **Mandatory section**

### Reviewers

(FILL ME IN) Reviewer 1 (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: test/DynamoCoreTests/CodeBlockNodeTests.cs [hunk 1]
```diff
@@ -1810,7 +1811,7 @@ public void ImportStatementInCodeBlock_DoesNotLoadAssemblyIntoProcess()
             string assemblyName = "FFITarget";
             UpdateCodeBlockNodeContent(codeBlockNode, $"import(\"{assemblyName}.dll\")");
 
-            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+            var loadedAssemblies = AssemblyLoadContext.All.SelectMany(context => context.Assemblies);
 
             var ffiTargetAsm = loadedAssemblies.Any(assembly => assembly.GetName().Name.Equals(assemblyName, StringComparison.OrdinalIgnoreCase));
```
Comment (aparajit-pratap) [inline test/DynamoCoreTests/CodeBlockNodeTests.cs:1817]: `ffiTargetAsm` is turning out to be `true` in Linux :(

File: test/DynamoCoreTests/CodeBlockNodeTests.cs [hunk 2]
```diff
@@ -1810,7 +1811,7 @@ public void ImportStatementInCodeBlock_DoesNotLoadAssemblyIntoProcess()
             string assemblyName = "FFITarget";
             UpdateCodeBlockNodeContent(codeBlockNode, $"import(\"{assemblyName}.dll\")");
 
-            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+            var loadedAssemblies = AssemblyLoadContext.All.SelectMany(context => context.Assemblies);
```
Comment (mjkkirschner) [inline test/DynamoCoreTests/CodeBlockNodeTests.cs:1814]: whats the difference between these? Is CurrentDomain pointing to just the default ALC?

**Comment by mjkkirschner** [comment]
also curious, where are the tests actually running on linux?

**Comment by aparajit-pratap** [comment]
> also curious, where are the tests actually running on linux?

They are running on `DynamoWinBuild` [here](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/net60_linux_target-release/).

**Review by mjkkirschner — APPROVED** [review]

**Comment by mjkkirschner** [comment]
I guess we will want to make sure this new assembly is not delivered as part of core in our various nugets and builds?

**Comment by aparajit-pratap** [comment]
> I guess we will want to make sure this new assembly is not delivered as part of core in our various nugets and builds?

Is there anything special we do for FFITarget?

**Comment by mjkkirschner** [comment]
> > I guess we will want to make sure this new assembly is not delivered as part of core in our various nugets and builds?
> 
> Is there anything special we do for FFITarget?

I think we actually inadvertently distribute FFITarget today - but for other test binaries we exclude them for the dynamocore builds on master-15 here:
https://github.com/DynamoDS/Dynamo/blob/f4210a1191c246270634cf28dd3a014baff7bef5/src/DynamoInstall/DynamoInstall.wixproj#L77

For the nugets I think we exclude in the nuspecs.

**Comment by aparajit-pratap** [comment]
@mjkkirschner I'll make a separate PR to exclude test related assemblies from the nuspec definition for the serviceruntimecore package.

**Review by mjkkirschner — APPROVED** [review]

**Pull request merged**